Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1026,55 +1030,138 @@ class BrowserViewController: UIViewController {
         
         super.updateViewConstraints()
     }
-
-    fileprivate func showHomePanelController(inline: Bool) {
-        homePanelIsInline = inline
-
-        if favoritesViewController == nil {
-            let homePanelController = FavoritesViewController(profile: profile,
-                                                              fromOverlay: !inline,
-                                                              rewards: rewards,
-                                                              backgroundDataSource: backgroundDataSource)
-            homePanelController.delegate = self
-            homePanelController.view.alpha = 0
-            homePanelController.applyTheme(Theme.of(tabManager.selectedTab))
-
-            self.favoritesViewController = homePanelController
-
-            addChild(homePanelController)
-            view.addSubview(homePanelController.view)
-            homePanelController.didMove(toParent: self)
+    
+    private(set) var favoritesController: FavoritesViewController?
+    
+    private func displayFavoritesController() {
+        if favoritesController == nil {
+            let favoritesController = FavoritesViewController { [weak self] bookmark, action in
+                self?.handleBookmarkAction(bookmark: bookmark, action: action)
+            }
+            favoritesController.applyTheme(Theme.of(tabManager.selectedTab))
+            self.favoritesController = favoritesController
+            
+            addChild(favoritesController)
+            view.addSubview(favoritesController.view)
+            favoritesController.didMove(toParent: self)
+            
+            favoritesController.view.snp.makeConstraints {
+                $0.top.leading.trailing.equalTo(pageOverlayLayoutGuide)
+                $0.bottom.equalTo(view)
+            }
+        }
+        guard let favoritesController = favoritesController else { return }
+        favoritesController.view.alpha = 0.0
+        let animator = UIViewPropertyAnimator(duration: 0.2, dampingRatio: 1.0) {
+            favoritesController.view.alpha = 1
+        }
+        animator.addCompletion { _ in
+            self.webViewContainer.accessibilityElementsHidden = true
+            UIAccessibility.post(notification: .screenChanged, argument: nil)
+        }
+        animator.startAnimation()
+    }
+    
+    private func hideFavoritesController() {
+        if let controller = favoritesController {
+            self.favoritesController = nil
+            UIView.animate(withDuration: 0.1, delay: 0, options: [.beginFromCurrentState], animations: {
+                controller.view.alpha = 0.0
+            }, completion: { _ in
+                controller.willMove(toParent: nil)
+                controller.view.removeFromSuperview()
+                controller.removeFromParent()
+                self.webViewContainer.accessibilityElementsHidden = false
+                UIAccessibility.post(notification: .screenChanged, argument: nil)
```

Comment:
I assume for accessibility reasons since we are not presenting via UIViewController chain (present/dismiss). I copy pasted this from original NTP code